### ISSUE TITLE
Bug in the default method for `IsInverseSemigroup`

### ISSUE THREAD

**Issue by wilfwilson** [gap-packages/Semigroups#377]
```
gap> S := Semigroup([
>   Transformation([1, 1, 1, 2, 1, 5, 3]),
>   Transformation([1, 4, 4, 4, 6, 6, 6])],
> rec(acting := false));;
gap> IsInverseSemigroup(S);
true
gap> S := Semigroup(S, rec(acting := true));;
gap> IsInverseSemigroup(S);
false
gap> S := Semigroup(S, rec(acting := true));;
gap> IsSemigroupWithCommutingIdempotents(S);
false
gap> IsRegularSemigroup(S);
false
```
The offending line is `692` in `gap/attributes/properties.gi`:
```
return NrLClasses(S) = NrRClasses(S) and NrIdempotents(S) = NrRClasses(S)
```
This condition is certainly necessary for an inverse semigroup, but it is not sufficient, as the above example shows. This can be sorted by additionally checking `IsRegularSemigroup(S)` in this line.

**Comment by james-d-mitchell** [comment]
Good catch, lots of bugs found today!

**Comment by james-d-mitchell** [comment]
Resolved by PR #378
